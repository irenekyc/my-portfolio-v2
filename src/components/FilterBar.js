import React, {useEffect, useRef} from 'react'
import {useSelector, useDispatch} from 'react-redux'
import {fetchData} from '../actions/data'


const FilterBar = ()=>{
    const dispatch = useDispatch()
    const {curPage, sort, filter, init}= useSelector(state=>state.status)
    const scrollToRef = useRef(null)

    useEffect(()=>{
        if(!init){
            window.scrollTo(0,scrollToRef.current.offsetTop-60)
        }
    }, [curPage])
    
    return(
        <section className="filter-session" ref={scrollToRef}>
        <div className="filter-session-inner">
            <div className="filter">
                <span> Filter by</span>
                <button className={filter === "SCSS" && `filter-active`} onClick={()=>dispatch(fetchData(0, sort, "SCSS"))}> Scss</button>
                <button className={filter === "JS" && `filter-active`} onClick={()=>dispatch(fetchData(0, sort, "JS"))}> JS</button>
                <button className={filter === "API" && `filter-active`}onClick={()=>dispatch(fetchData(0, sort, "API"))}> API</button>
                <button className={filter === "react" && `filter-active`} onClick={()=>dispatch(fetchData(0, sort, "react"))}> React js</button>
                <button className={filter === "MongoDB" && `filter-active`} onClick={()=>dispatch(fetchData(0, sort, "MongoDB"))}> MongoDB </button>
                <button onClick={()=>dispatch(fetchData(0, sort, "none"))}> Clear</button>
            </div>

            <div className="sort">
                <span> Sort by</span>
            <button className={sort === "desc" && `sort-active`} onClick={()=>dispatch(fetchData(0, "desc", filter))}> Latest</button>
            <button className={sort === "asc" && `sort-active`} onClick={()=>dispatch(fetchData(0, "asc", filter))}> Earliest</button>
            </div>
          
        </div>

      

    </section>
    )
}

export default FilterBar