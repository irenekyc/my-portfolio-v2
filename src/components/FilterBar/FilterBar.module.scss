.ikFilterBar {
  &__container {
    border: 1px solid #00adb5;
    color: #00adb5;
    background: #3a4750 !important;
    padding: 0.5rem 0;
    position: sticky;
    top: 68px;
    width: 100%;
  }
  &__inner {
    height: 100%;
    width: 80%;
    max-width: 900px;
    margin-right: auto;
    margin-left: auto;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    flex-direction: column;
    -webkit-box-pack: justify;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    button {
      padding: 0.25rem 0.75rem;
      font-size: 1rem;
      font-family: inherit;
      border: none;
      outline: none;
      color: #eee;
      background: transparent;
      cursor: pointer;
      &:hover {
        color: #00adb5;
      }
    }
  }
  &__button__active {
    color: #00adb5 !important;
  }
}

// .portfolio-container {

// }

// .portfolio-container .portfolio-card * {
//   margin-bottom: 0.5rem;
// }
// .portfolio-container .portfolio-card a {

// }
// .portfolio-container .portfolio-card a:hover {
//   color: #00adb5;
// }
// .portfolio-container .portfolio-card-heading {
//   padding: 0 1rem;
//   display: -webkit-box;
//   display: -ms-flexbox;
//   display: flex;
//   -webkit-box-pack: justify;
//   -ms-flex-pack: justify;
//   justify-content: space-between;
//   -webkit-box-align: end;
//   -ms-flex-align: end;
//   align-items: flex-end;
// }
// .portfolio-container .portfolio-card-heading-main {
//   font-size: 2.5rem;
// }
// .portfolio-container .portfolio-card .portoflio-card-content {
//   -webkit-box-flex: 1;
//   -ms-flex: 1;
//   flex: 1;
// }
// .portfolio-container .portfolio-card button {
//   border: 1px solid #303841;
//   background: transparent;
//   padding: 4px 8px;
//   outline: none;
//   cursor: pointer;
// }
// .portfolio-container .portfolio-card button:hover {
//   color: #00adb5;
//   border: 1px solid #00adb5;
// }
// footer {
//   text-align: center;
// }
// @media (min-width: 766px) {
//   .hero-banner .hero-banner-inner {
//     -ms-grid-columns: 1fr 1fr 1fr;
//     grid-template-columns: 1fr 1fr 1fr;
//     grid-template-areas: "card1 card2 card3";
//   }
//   .hero-banner #coding-hours {
//     -ms-flex-item-align: unset;
//     -ms-grid-row-align: unset;
//     align-self: unset;
//     margin-right: unset;
//     margin-left: unset;
//   }
//   .modal-menu {
//     -webkit-box-orient: horizontal;
//     -webkit-box-direction: normal;
//     -ms-flex-direction: row;
//     flex-direction: row;
//   }
// }

// .featured-heading i {
//   margin-right: 1rem;
//   font-size: 2rem;
// }

// .filter-session-inner .filter,
// .filter-session-inner .sort {
//   width: 100%;
//   max-width: 600px;
//   display: flex;
//   justify-content: space-between;
//   align-items: center;
//   flex-wrap: wrap;
//   padding: 0.2rem 1rem;
// }

// .filter-session-inner .filter button {
//   margin: 0.5rem 0;
// }

// @media (min-width: 900px) {
//   .filter-session-inner {
//     flex-direction: row;
//   }

//   .filter-session-inner .filter {
//     width: 70%;
//     padding: 0.5rem;
//   }
//   .filter-session-inner .sort {
//     width: 30%;
//     padding: 0.5rem;
//   }
// }
