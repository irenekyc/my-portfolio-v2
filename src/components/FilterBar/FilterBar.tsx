import React, { useState, FunctionComponent } from "react";
import styles from "./FilterBar.module.scss";

interface FilterBarProps {
  setConditions: ({
    filter,
    sort,
  }: {
    filter: string | null;
    sort: string;
  }) => void;
}

const FilterBar: FunctionComponent<FilterBarProps> = ({
  setConditions,
}: FilterBarProps) => {
  const [filter, setFilter] = useState<string | null>(null);
  const [sort, setSort] = useState<string>("desc");
  const onClickFilter = (e: any) => {
    const id = e.target.id;
    setFilter(id);
    setConditions({
      sort,
      filter: id,
    });
  };

  const onClickSort = (e: any) => {
    const id = e.target.id;
    setSort(id);
    setConditions({
      sort: id,
      filter,
    });
  };

  const clearFilter = () => {
    setConditions({
      sort,
      filter: null,
    });
    setFilter(null);
  };

  return (
    <section
      className={styles.ikFilterBar__container}
      id="filter-bar"
      data-testid="filter-bar"
    >
      <div className={styles.ikFilterBar__inner}>
        <div className="filter">
          <span> Filter by</span>
          <button
            className={
              filter === "Next" ? styles.ikFilterBar__button__active : ""
            }
            id={"Next"}
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            NextJs
          </button>
          <button
            className={
              filter === "JS" ? styles.ikFilterBar__button__active : ""
            }
            id={"JS"}
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            JS
          </button>
          <button
            className={
              filter === "API" ? styles.ikFilterBar__button__active : ""
            }
            id="API"
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            API
          </button>
          <button
            className={
              filter === "React" ? styles.ikFilterBar__button__active : ""
            }
            id="React"
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            React
          </button>
          <button
            className={
              filter === "MongoDB" ? styles.ikFilterBar__button__active : ""
            }
            id="MongoDB"
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            MongoDB{" "}
          </button>
          <button
            className={
              filter === "Solidity" ? styles.ikFilterBar__button__active : ""
            }
            id="Solidity"
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            Solidity{" "}
          </button>
          <button
            className={
              filter === "Python" ? styles.ikFilterBar__button__active : ""
            }
            id="Python"
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            Python{" "}
          </button>
          <button
            className={
              filter === "Webflow" ? styles.ikFilterBar__button__active : ""
            }
            id="Webflow"
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            Webflow{" "}
          </button>
          <button
            className={
              filter === "Typescript" ? styles.ikFilterBar__button__active : ""
            }
            id="Typescript"
            onClick={onClickFilter}
            data-testid="filter-option"
          >
            {" "}
            Typescript{" "}
          </button>
          <button onClick={clearFilter}> Clear</button>
        </div>

        <div className="sort">
          <span> Sort by</span>
          <button
            className={
              sort === "desc" ? styles.ikFilterBar__button__active : ""
            }
            id="desc"
            onClick={onClickSort}
            data-testid="sort-option"
          >
            {" "}
            Latest
          </button>
          <button
            className={sort === "asc" ? styles.ikFilterBar__button__active : ""}
            id="asc"
            onClick={onClickSort}
            data-testid="sort-option"
          >
            {" "}
            Earliest
          </button>
        </div>
      </div>
    </section>
  );
};

export default FilterBar;
