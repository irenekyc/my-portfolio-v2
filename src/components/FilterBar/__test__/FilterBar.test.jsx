import React from "react";
import FilterBar from "../index";
import { render, screen, fireEvent } from "@testing-library/react";

const buttonActiveClass = "ikFilterBar__button__active";

it("Render Filter Bar", () => {
  const mockSetConditions = jest.fn();
  render(<FilterBar setConditions={mockSetConditions} />);
  const filterBarSection = screen.getByTestId("filter-bar");
  expect(filterBarSection).toBeInTheDocument();
  const filterBarOptions = screen.getAllByTestId("filter-option");
  expect(filterBarOptions).toHaveLength(9);
  const sortOptions = screen.getAllByTestId("sort-option");
  expect(sortOptions).toHaveLength(2);
});

it("Render Filter Bar - sort by latest option should be default", () => {
  const mockSetConditions = jest.fn();
  render(<FilterBar setConditions={mockSetConditions} />);
  const filterBarSection = screen.getByTestId("filter-bar");
  expect(filterBarSection).toBeInTheDocument();
  const latestOption = screen.getByText("Latest");
  expect(latestOption).toBeInTheDocument();
  expect(latestOption).toHaveClass(buttonActiveClass);
});

it("Click on filter option - the button should be active and set conditions to be called correctly", () => {
  const mockSetConditions = jest.fn();
  render(<FilterBar setConditions={mockSetConditions} />);
  const filterBarOptions = screen.getAllByTestId("filter-option");
  expect(filterBarOptions[0]).not.toHaveClass(buttonActiveClass);
  fireEvent.click(filterBarOptions[0], { bubble: true, cancellable: true });
  expect(filterBarOptions[0]).toHaveClass(buttonActiveClass);
  expect(mockSetConditions).toBeCalledWith({ filter: "Next", sort: "desc" });
});

it("Click on sort option - the button should be active and set conditions to be called correctly", () => {
  const mockSetConditions = jest.fn();
  render(<FilterBar setConditions={mockSetConditions} />);

  const earliestOption = screen.getByText("Earliest");
  expect(earliestOption).not.toHaveClass(buttonActiveClass);
  fireEvent.click(earliestOption);
  expect(earliestOption).toHaveClass(buttonActiveClass);

  expect(mockSetConditions).toBeCalledWith({ filter: null, sort: "asc" });
});
