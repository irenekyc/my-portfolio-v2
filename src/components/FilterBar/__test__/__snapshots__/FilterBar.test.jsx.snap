// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Filter Bar 1`] = `
<section
  class="ikFilterBar__container"
  data-testid="filter-bar"
  id="filter-bar"
>
  <div
    class="ikFilterBar__inner"
  >
    <div
      class="filter"
    >
      <span>
         Filter by
      </span>
      <button
        class=""
        data-testid="filter-option"
        id="Next"
      >
         
        NextJs
      </button>
      <button
        class=""
        data-testid="filter-option"
        id="JS"
      >
         
        JS
      </button>
      <button
        class=""
        data-testid="filter-option"
        id="API"
      >
         
        API
      </button>
      <button
        class=""
        data-testid="filter-option"
        id="React"
      >
         
        React
      </button>
      <button
        class=""
        data-testid="filter-option"
        id="MongoDB"
      >
         
        MongoDB
         
      </button>
      <button
        class=""
        data-testid="filter-option"
        id="Solidity"
      >
         
        Solidity
         
      </button>
      <button
        class=""
        data-testid="filter-option"
        id="Python"
      >
         
        Python
         
      </button>
      <button
        class=""
        data-testid="filter-option"
        id="Webflow"
      >
         
        Webflow
         
      </button>
      <button
        class=""
        data-testid="filter-option"
        id="Typescript"
      >
         
        Typescript
         
      </button>
      <button>
         Clear
      </button>
    </div>
    <div
      class="sort"
    >
      <span>
         Sort by
      </span>
      <button
        class="ikFilterBar__button__active"
        data-testid="sort-option"
        id="desc"
      >
         
        Latest
      </button>
      <button
        class=""
        data-testid="sort-option"
        id="asc"
      >
         
        Earliest
      </button>
    </div>
  </div>
</section>
`;
